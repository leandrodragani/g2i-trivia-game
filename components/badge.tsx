import React, { useContext } from "react";
import { ThemeContext } from "styled-components/native";
import { Text } from "./text";
import { Box } from "./box";

export interface BadgeProps {
  color: string;
  label: string;
}

export function Badge({ color, label }: BadgeProps) {
  const theme = useContext(ThemeContext);
  return (
    <Box bg={color} borderRadius={4} paddingX={2} paddingY={1}>
      <Text
        color={theme.colors.white}
        fontFamily={theme.font.medium}
        fontSize={14}
        style={{ textTransform: "uppercase" }}
        numberOfLines={1}
        ellipsizeMode="tail"
      >
        {label}
      </Text>
    </Box>
  );
}
