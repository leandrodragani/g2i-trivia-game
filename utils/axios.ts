import axios from "axios";

export const apiInstance = axios.create({
  baseURL: "https://opentdb.com",
});

export const fetcher = (url: string) =>
  apiInstance.get(url).then((res) => res.data);
