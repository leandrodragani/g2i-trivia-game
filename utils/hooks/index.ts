export * from "./use-cached-resources";
export * from "./use-modal";
export * from "./use-before-leave";