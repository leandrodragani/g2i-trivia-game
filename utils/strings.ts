export function objectToQueryParams(object: any) {
  const sanitize = (object: any) => {
    Object.keys(object).forEach((key) =>
      object[key] === undefined || object[key] === null
        ? delete object[key]
        : {}
    );
    return object;
  };

  const queryString = new URLSearchParams(sanitize(object)).toString();
  return queryString;
}
