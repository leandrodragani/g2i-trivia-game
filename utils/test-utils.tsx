import React from "react";
import { NavigationContainer } from "@react-navigation/native";
import { render, RenderOptions } from "@testing-library/react-native";
import { ThemeProvider } from "styled-components/native";
import theme from "styles/theme";
import { ToastProvider } from "context";
import { SafeAreaProvider } from "react-native-safe-area-context";

interface WrappedProvidersProps {
  children: React.ReactNode;
}

const WrappedProviders = ({ children }: WrappedProvidersProps) => {
  return (
    <ThemeProvider {...{ theme }}>
      <SafeAreaProvider>
        <NavigationContainer>
          <ToastProvider>{children}</ToastProvider>
        </NavigationContainer>
      </SafeAreaProvider>
    </ThemeProvider>
  );
};

const customRender = (ui: React.ReactElement<any>, options?: RenderOptions) =>
  render(ui, { wrapper: WrappedProviders, ...options });

export * from "@testing-library/react-native";

export { customRender as render };
