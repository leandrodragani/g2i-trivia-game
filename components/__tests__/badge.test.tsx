import React from "react";
import { render } from "utils/test-utils";
import { Badge } from "../badge";

test("must show correctly", async () => {
  const { toJSON, findByText } = render(<Badge label="Badge" color="red" />);
  expect(findByText("Badge")).toBeTruthy();
  expect(toJSON).toMatchSnapshot();
});
