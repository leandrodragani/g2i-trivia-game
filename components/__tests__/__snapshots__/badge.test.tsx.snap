// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must show correctly 1`] = `[Function]`;

exports[`shows correctly 1`] = `[Function]`;
